### PR TITLE
[FrontDoor] New-AzFrontDoorRulesEngineActionObject: Allow rule engine action creation without RouteConfigurationOverride and fix DynamicCompression parameter being ignored issue

### PR DESCRIPTION
**Pull request by t-bzhan**
Fix multiple issues in New-AzFrontDoorRulesEngineActionObject cmdlet

## Description

- Allow rule engine action creation without RouteConfigurationOverride 
- Fix DynamicCompression parameter being ignored issue

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: tools/StaticAnalysis/Exceptions/Az.FrontDoor/SignatureIssues.csv [hunk 1]
```diff
@@ -18,3 +18,4 @@
...
 "Microsoft.Azure.Commands.FrontDoor.dll","Microsoft.Azure.Commands.FrontDoor.Cmdlets.NewFrontDoorRulesEngineRuleObject","New-AzFrontDoorRulesEngineRuleObject","1","8100","New-AzFrontDoorRulesEngineRuleObject Does not support ShouldProcess but the cmdlet verb New indicates that it should.","Determine if the cmdlet should implement ShouldProcess and if so determine if it should implement Force / ShouldContinue"
+"Az.FrontDoor","Microsoft.Azure.Commands.FrontDoor.Cmdlets.NewFrontDoorRulesEngineActionObject","New-AzFrontDoorRulesEngineActionObject","1","8700","Parameter set '__AllParameterSets', 'ByFieldsWithRedirectParameterSet' of cmdlet 'New-AzFrontDoorRulesEngineActionObject' have the same mandatory parameters, and both of them are not default parameter set which may cause confusion.","Merge these parameter sets into one parameter set."
```
Comment (t-bzhan) [inline tools/StaticAnalysis/Exceptions/Az.FrontDoor/SignatureIssues.csv:21]: That sounds like a StaticAnalysis issue, as I've already mark ByFieldsWithRegularActionParameterSet as the default parameter set.

File: src/FrontDoor/FrontDoor/Cmdlets/NewFrontDoorRulesEngineActionObject.cs [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
 namespace Microsoft.Azure.Commands.FrontDoor.Cmdlets
 {
-    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "FrontDoor" + "RulesEngineActionObject"), OutputType(typeof(PSRulesEngineAction))]
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "FrontDoor" + "RulesEngineActionObject", DefaultParameterSetName = FieldsWithRegularActionParameterSet), OutputType(typeof(PSRulesEngineAction))]
```
Comment (wyunchi-ms) [inline src/FrontDoor/FrontDoor/Cmdlets/NewFrontDoorRulesEngineActionObject.cs:10]: ParameterSet `FieldsWithRegularActionParameterSet` is not defined

Comment (t-bzhan) [reply src/FrontDoor/FrontDoor/Cmdlets/NewFrontDoorRulesEngineActionObject.cs:10]: Hi @wyunchi-ms , I followed the convention to add a public field in https://github.com/Azure/azure-powershell/pull/15852/files#diff-faf23f354d9f562c6e2599390b451bb1a693513998eb34dc788143a917b575bc:
`public const string FieldsWithRegularActionParameterSet = "ByFieldsWithRegularActionParameterSet";`

Is there other places I did not take care of for this parameter set?

Comment (t-bzhan) [reply src/FrontDoor/FrontDoor/Cmdlets/NewFrontDoorRulesEngineActionObject.cs:10]: I suppose this ParameterSet will be applied implicitly to all Parameters properties without  ParameterSetName explicitly defined in NewFrontDoorRulesEngineActionObject.cs

Comment (wyunchi-ms) [reply src/FrontDoor/FrontDoor/Cmdlets/NewFrontDoorRulesEngineActionObject.cs:10]: Sorry, I mean `FieldsWithRegularActionParameterSet` is not referenced in the parameter definations.

**Review by wyunchi-ms — APPROVED** [review]

**Pull request merged**